subprojects/wrapper/src/integTest/groovy/org/gradle/integtests/WrapperHttpsIntegrationTest.groovy

```diff
@@ -71,4 +75,51 @@ class WrapperHttpsIntegrationTest extends AbstractWrapperIntegrationSpec {
         then:
         outputDoesNotContain('WARNING Using HTTP Basic Authentication over an insecure connection to download the Gradle distribution. Please consider using HTTPS.')
     }
+
+    def "downloads wrapper via https proxy"() {
+        given:
+        proxyServer.start()
+        prepareWrapper(server.uri.toString())
+        file("gradle.properties") << """
+    systemProp.https.proxyHost=localhost
+    systemProp.https.proxyPort=${proxyServer.port}
+    systemProp.https.nonProxyHosts=${JavaVersion.current() >= JavaVersion.VERSION_1_7 ? '' : '~localhost'}
+"""
+        server.expect(server.get("/gradlew/dist").sendFile(distribution.binDistribution))
+
+        when:
+        def result = wrapperExecuter.withTasks('hello').run()
+
+        then:
+        assertThat(result.output, containsString('hello'))
+
+        and:
+        proxyServer.requestCount == 1
+    }
+
+    @Issue('https://github.com/gradle/gradle/issues/5052')
+    def "downloads wrapper via authenticated https proxy"() {
+        given:
+        proxyServer.start('my_user', 'my_password')
+
+        and:
+        prepareWrapper(server.uri.toString())
+        server.expect(server.get("/gradlew/dist").sendFile(distribution.binDistribution))
+        file("gradle.properties") << """
+    systemProp.https.proxyHost=localhost
+    systemProp.https.proxyPort=${proxyServer.port}
+    systemProp.https.nonProxyHosts=${JavaVersion.current() >= JavaVersion.VERSION_1_7 ? '' : '~localhost'}
```

Comment:
❌ The Gradle tests are expected to run on Java 1.8 and above. You can remove the branch here.

Reply:
I copied this from the existing `WrapperHttpIntegrationTest`. Should I remove this check also from there?

Reply:
I just did that.

Reply:
I think this was actually doing something important.  By default, Java won't use a proxy for localhost (`nonProxyHosts` contains localhost by default).  This test was clearing `nonProxyHosts` so that the proxy will be used.  We still need to do that

Reply:
Ah, so the condition does not intend to treat Java >= 1.7 specially, but Java < 1.7 specially. I'll drop the commit and adjust my changes again.